setup.py

```diff
@@ -35,14 +37,91 @@ def is_conda_interpreter():
     """
     return 'conda' in sys.version or 'Continuum' in sys.version
 
+def is_macOS():
+    return 'darwin' in sys.platform
+
+
+# HOMEBREW_SAMPLE = """$ brew ls --verbose libomp
+# /opt/homebrew/Cellar/libomp/15.0.7/INSTALL_RECEIPT.json
+# /opt/homebrew/Cellar/libomp/15.0.7/.brew/libomp.rb
+# /opt/homebrew/Cellar/libomp/15.0.7/include/ompt.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp-tools.h
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.dylib
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.a
+# """
+#
+# MACPORTS_SAMPLE = """$ port contents libomp
+# Port libomp contains:
+#   /opt/local/include/libomp/omp-tools.h
+#   /opt/local/include/libomp/omp.h
+#   /opt/local/include/libomp/ompt.h
+#   /opt/local/lib/libomp/libgomp.dylib
+#   /opt/local/lib/libomp/libiomp5.dylib
+#   /opt/local/lib/libomp/libomp.dylib
+#   /opt/local/share/doc/libomp/LICENSE.TXT
+#   /opt/local/share/doc/libomp/README.txt
+# """
+
+
+def compile_link_paths_from_manifest(cmd):
+    from subprocess import run
+    query = run(cmd, shell=True, check=False, capture_output=True)
+    if query.returncode != 0:
+        return None, None
+    manifest = query.stdout.decode("UTF-8")
+    # find all the unique directories mentioned in the manifest
+    dirs = set(os.path.split(filename)[0] for filename in re.findall(r'^\s*(/.*?)\s*$', manifest, re.MULTILINE))
+    # find a unique libdir and incdir
+    inc = tuple(d for d in dirs if re.search(r'/include(\W|$)', d))
+    lib = tuple(d for d in dirs if re.search(r'/lib(\W|$)', d))
+    # only return success if there's no ambiguity
+    return (inc + lib) if len(inc) == 1 and len(lib) == 1 else (None, None)
+
+
+@lru_cache(maxsize=1)
+def macOS_omp_options_from_probe():
+    """Get common include and library paths for libomp installation on macOS.
+       For example ``(['-I/opt/local/include/libomp'], ['-L/opt/local/lib/libomp'])``.
+    """
+    if is_macOS():
+        for cmd in ["brew ls --verbose libomp", "port contents libomp"]:
+            inc, lib = compile_link_paths_from_manifest(cmd)
+            if inc and lib:
+                return [f"-I{inc}"], [f"-L{lib}"]
+    return [], []
+
+def is_macOS_with_libomp():
+    if not is_macOS():
+        return False
+    inc, lib = macOS_omp_options_from_probe()
+    return bool(inc) and bool(lib)
+
+OMP_TAB = {
+    '1': 'gomp',
```

Comment:
Shouldn't this be `'probe'`? Or I mean the same as `probe`? Exact backwards compatibility might be `gomp` but wouldn't that be the same end result as `probe` for the systems that this logic already worked on?